src/librustc_codegen_llvm/llvm_util.rs

```diff
@@ -257,8 +257,7 @@ pub fn target_feature_whitelist(sess: &Session)
         "hexagon" => HEXAGON_WHITELIST,
         "mips" | "mips64" => MIPS_WHITELIST,
         "powerpc" | "powerpc64" => POWERPC_WHITELIST,
-        // wasm32 on emscripten does not support these target features
-        "wasm32" if !sess.target.target.options.is_like_emscripten => WASM_WHITELIST,
+        "wasm32" => WASM_WHITELIST,
```

Comment:
I noticed that most of the recent features are not available in rustc. What is the policy for adding the latest and greatest feature flags?

Reply:
This list here actually only corresponds to `#[cfg(target_feature = "..")]` in code itself, you can always pass any set of flags to `-C target-feature=...` that LLVM supports. In that sense there's not too much use in filling this out for wasm since most of them you basically need to recompiled the whole crate graph and it's for minor features like bulk-memory that code doesn't typically need to read.

All that to say though that we can fill it out at any time, I think the use case motivating this otherwise just hasn't come up yet!